src/renderer/api/endpoints/helm-charts.api.ts

```diff
@@ -83,81 +85,303 @@ export async function getChartValues(repo: string, name: string, version: string
+      stripUnknown: {
+        arrays: true
+      },
+    })
+    .default(() => ([])),
+  urls: Joi
+    .array()
+    .items(Joi.string())
+    .options({
+      stripUnknown: {
+        arrays: true
+      },
+    })
+    .default(() => ([])),
+  maintainers: Joi
+    .array()
+    .items(helmChartMaintainerValidator)
+    .options({
+      stripUnknown: {
+        arrays: true
+      },
+    })
+    .default(() => ([])),
+  dependencies: Joi
+    .array()
+    .items(helmChartDependencyValidator)
+    .options({
+      stripUnknown: {
+        arrays: true
+      },
+    })
+    .default(() => ([])),
+  annotations: Joi
+    .object({})
+    .pattern(/.*/, Joi.string())
+    .default(() => ({})),
+});
+
+export interface HelmChartCreateOpts {
+  onError?: "throw" | "log";
+}
+
+export interface HelmChartMaintainer {
+  name: string;
+  email: string;
+  url?: string;
+}
+
+export interface RawHelmChartDependency {
+  name: string;
+  repository: string;
+  condition?: string;
+  version: string;
+  tags?: string[];
+}
+
+export type HelmChartDependency = Required<Omit<RawHelmChartDependency, "condition">>
+  & Pick<RawHelmChartDependency, "condition">;
+
 export interface HelmChart {
   apiVersion: string;
   name: string;
   version: string;
   repo: string;
   kubeVersion?: string;
   created: string;
-  description?: string;
+  description: string;
   digest: string;
-  keywords?: string[];
+  keywords: string[];
   home?: string;
-  sources?: string[];
-  maintainers?: {
-    name: string;
-    email: string;
-    url: string;
-  }[];
+  sources: string[];
+  urls: string[];
+  annotations: Record<string, string>;
+  dependencies: HelmChartDependency[];
+  maintainers: HelmChartMaintainer[];
   engine?: string;
   icon?: string;
   appVersion?: string;
-  deprecated?: boolean;
+  type?: string;
+  deprecated: boolean;
   tillerVersion?: string;
 }
 
 export class HelmChart {
-  constructor(data: HelmChart) {
-    Object.assign(this, data);
+  private constructor(value: HelmChart) {
+    this.apiVersion = value.apiVersion;
+    this.name = value.name;
+    this.version = value.version;
+    this.repo = value.repo;
+    this.kubeVersion = value.kubeVersion;
+    this.created = value.created;
+    this.description = value.description;
+    this.digest = value.digest;
+    this.keywords = value.keywords;
+    this.home = value.home;
+    this.sources = value.sources;
+    this.maintainers = value.maintainers;
+    this.engine = value.engine;
+    this.icon = value.icon;
+    this.apiVersion = value.apiVersion;
+    this.deprecated = value.deprecated;
+    this.tillerVersion = value.tillerVersion;
+    this.annotations = value.annotations;
+    this.urls = value.urls;
+    this.dependencies = value.dependencies;
+    this.type = value.type;
+
     autoBind(this);
   }
 
-  static create(data: any) {
-    return new HelmChart(data);
+  static create(data: RawHelmChart, { onError = "throw" }: HelmChartCreateOpts = {}): HelmChart | undefined {
```

Comment:
since we are not handling onError == "log", should we use `{ throw: true }` instead?

Reply:
Sure, I meant the `console.warn` on line 339 to be the handling of `onError = "log"`

Reply:
yes it does, and it handles also `onError="foo"`. So, since only `onError="throw"` has effect, it sounds more like boolean value, so passing `{ throw:true|false }` would be more logical. But this is only semantics